apps/devhub/tasks.py

```diff
@@ -32,65 +32,116 @@
 log = logging.getLogger('z.devhub.task')
 
 
+# Simple wrapper for the sake of tests.
+def validate(file_, listed=None):
```

Comment:
Did you write `listed=None` instead of `listed=True` on purpose? It feels weird to have `None` for something that we expect to be boolean. I see you've reproduced this in quite a few places along the way.